tests/xcm-simulator/src/tests.rs

```diff
@@ -16,13 +16,53 @@
 // You should have received a copy of the GNU General Public License
 // along with Astar. If not, see <http://www.gnu.org/licenses/>.
 
-use crate::mocks::{parachain, relay_chain, *};
+use crate::mocks::{msg_queue::mock_msg_queue, parachain, relay_chain, *};
 
-use frame_support::{assert_ok, weights::Weight};
+use frame_support::{assert_ok, traits::IsType, weights::Weight};
 use parity_scale_codec::Encode;
-use xcm::latest::prelude::*;
+use sp_runtime::{
+    traits::{Bounded, StaticLookup},
+    DispatchResult,
+};
+use xcm::prelude::*;
 use xcm_simulator::TestExt;
 
+fn register_asset<Runtime, AssetId>(
```

Comment:
> Should this be moved to some utils file?

yes, I'll move it to utils.rs along with contract related helper functions

> That's a lot of arguments - why not improve it a bit by passing AssetIdParameter instead?

Unfortuently they all those arguments are needed. 
Do you mean instead of using `AssetId`, use `AssetIdParameter`? how will that help in reducing args count?

Reply:
It shoudl help because you can put `min_balance`, `admin` & `is_sufficient` under it.

Reply:
https://github.com/AstarNetwork/Astar/blob/57419eee2f258d4735f96cdd03bc032668da79ca/tests/xcm-simulator/src/mocks/parachain.rs#L123-L124

I still don't understand, they both are same 😓

Reply:
You're right, my bad, sorry!

Not sure where, but I saw a struct with these 4 fields, hence the suggestion :see_no_evil: